### PR TITLE
feat(plasma-web/ui): Handling the onSearch event on pressing Enter

### PR DESCRIPTION
**Pull request by Yakutoc**
Необходимо добавить для компонентов TextField и Input обработку события по нажатию клавиши Enter. 

По [мотивам ant-design Input](https://github.com/ant-design/ant-design/blob/master/components/input/Search.tsx#L72-L77)

**Раньше:** 

Пользователю приходилось писать каждый раз **свою реализацию** такого обработчика.

**Стало:** 

```jsx
<TextField value={value} onSearch={onSearch} />
```

Снимок экрана 2023-01-28 в 12 54 01
     

:baby_chick: Download canary assets:

[colors--canary.294.4060402146.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.294.4060402146.xml)  
[color_metro_ios-swift--canary.294.4060402146.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.294.4060402146.swift)  
[color_metro_kotlin--canary.294.4060402146.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.294.4060402146.kt)  
[color_metro_react-native--canary.294.4060402146.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.294.4060402146.ts)  
[color_sbermarket_ios-swift--canary.294.4060402146.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.294.4060402146.swift)  
[color_sbermarket_kotlin--canary.294.4060402146.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.294.4060402146.kt)  
[color_sbermarket_react-native--canary.294.4060402146.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.294.4060402146.ts)  
[color_sberprime_ios-swift--canary.294.4060402146.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.294.4060402146.swift)  
[color_sberprime_kotlin--canary.294.4060402146.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.294.4060402146.kt)  
[color_sberprime_react-native--canary.294.4060402146.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.294.4060402146.ts)  
[color_selgros_ios-swift--canary.294.4060402146.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.294.4060402146.swift)  
[color_selgros_kotlin--canary.294.4060402146.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.294.4060402146.kt)  
[color_selgros_react-native--canary.294.4060402146.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.294.4060402146.ts)  
[color_smbusiness_ios-swift--canary.294.4060402146.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.294.4060402146.swift)  
[color_smbusiness_kotlin--canary.294.4060402146.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.294.4060402146.kt)  
[color_smbusiness_react-native--canary.294.4060402146.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.294.4060402146.ts)  
[PlasmaTokensColor--canary.294.4060402146.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.294.4060402146.swift)  
[typo_mage_ios-swift--canary.294.4060402146.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.294.4060402146.swift)  
[typo_mage_kotlin--canary.294.4060402146.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.294.4060402146.kt)  
[typo_mage_react-native--canary.294.4060402146.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.294.4060402146.ts)  
[typo_plasma_ios-swift--canary.294.4060402146.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.294.4060402146.swift)  
[typo_plasma_kotlin--canary.294.4060402146.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.294.4060402146.kt)  
[typo_plasma_react-native--canary.294.4060402146.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.294.4060402146.ts)  
[typo_ruler_ios-swift--canary.294.4060402146.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.294.4060402146.swift)  
[typo_ruler_kotlin--canary.294.4060402146.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.294.4060402146.kt)  
[typo_ruler_react-native--canary.294.4060402146.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.294.4060402146.ts)  
[typo_sage_ios-swift--canary.294.4060402146.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.294.4060402146.swift)  
[typo_sage_kotlin--canary.294.4060402146.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.294.4060402146.kt)  
[typo_sage_react-native--canary.294.4060402146.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.294.4060402146.ts)  
[typo_sbermarket_ios-swift--canary.294.4060402146.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.294.4060402146.swift)  
[typo_sbermarket_kotlin--canary.294.4060402146.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.294.4060402146.kt)  
[typo_sbermarket_react-native--canary.294.4060402146.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.294.4060402146.ts)  
[typo_soulmate_ios-swift--canary.294.4060402146.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.294.4060402146.swift)  
[typo_soulmate_kotlin--canary.294.4060402146.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.294.4060402146.kt)  
[typo_soulmate_react-native--canary.294.4060402146.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.294.4060402146.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.130.0-canary.294.4060402146.0
  npm install @salutejs/plasma-temple@1.127.0-canary.294.4060402146.0
  npm install @salutejs/plasma-ui@1.160.0-canary.294.4060402146.0
  npm install @salutejs/plasma-web@1.157.0-canary.294.4060402146.0
  # or 
  yarn add @salutejs/plasma-b2c@1.130.0-canary.294.4060402146.0
  yarn add @salutejs/plasma-temple@1.127.0-canary.294.4060402146.0
  yarn add @salutejs/plasma-ui@1.160.0-canary.294.4060402146.0
  yarn add @salutejs/plasma-web@1.157.0-canary.294.4060402146.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-294/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-294/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-294/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🔴 Fail**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-294/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-294/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-294/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-294/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-294/temple-storybook/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-294/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🔴 Fail**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-294/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-294/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-294/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-294/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-294/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-294/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-294/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🔴 Fail**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-294/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-294/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-294/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-294/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-294/temple-storybook/

File: packages/plasma-b2c/src/components/TextField/TextField.tsx [hunk 1]
```diff
@@ -81,6 +87,16 @@ export const TextField = forwardRef<HTMLInputElement, TextFieldProps>(function T
         [onChange],
     );
 
+    const handleSearch = useCallback(
+        (event: KeyboardEvent<HTMLInputElement>) => {
+            // INFO: Catch pressing the "Enter" key
+            if (event.keyCode === 13 && onSearch) {
+                onSearch((event.target as HTMLInputElement).value, event);
```
[the remark this answers is not in the document]
Comment (neretin-trike) [reply packages/plasma-b2c/src/components/TextField/TextField.tsx:94]: уловил, ts почему-то не может тут в типизацию

File: packages/plasma-web/src/components/Input/Input.tsx [hunk 2]
```diff
@@ -266,6 +272,16 @@ export const Input = forwardRef<HTMLInputElement, InputProps>(function Input(
         [onChange],
     );
 
+    const handleOnSearch = useCallback(
+        (event: KeyboardEvent<HTMLInputElement>) => {
+            // INFO: Catch pressing the "Enter" key
```
Comment (neretin-trike) [inline packages/plasma-web/src/components/Input/Input.tsx:277]: кажется, что комментарий тут не сильно нужен, т.к. у тебя в интерфейсе есть описание этого колбэка

Comment (Yakutoc) [reply packages/plasma-web/src/components/Input/Input.tsx:277]: @neretin-trike Хорошо, уберу.)

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-294/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-294/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-294/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-294/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-294/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-294/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-294/temple-storybook/

**Review by neretin-trike — APPROVED** [review]

File: packages/plasma-b2c/src/components/TextField/TextField.tsx [hunk 3]
```diff
@@ -106,6 +121,7 @@ export const TextField = forwardRef<HTMLInputElement, TextFieldProps>(function T
                     aria-labelledby={currentId}
                     aria-describedby={ariaDescribedBy}
                     onChange={handleChange}
+                    onKeyUp={handleSearch}
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/TextField/TextField.tsx:124]: а если мы напишем:  `onKeyUp={onSearch && handleSearch}` то это позволит нам слушать события только если передали обработчик?

**Review by Yeti-or — COMMENTED** [review]
что-то печально конечно что это надо в три места добавлять =(

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-294/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-294/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-294/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-294/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-294/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-294/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-294/temple-storybook/

**Review by Yeti-or — APPROVED** [review]

**Review by neretin-trike — APPROVED** [review]

**Pull request merged**